UI/MapUI.gd

```diff
@@ -18,6 +24,59 @@ func _input(_inputevent: InputEvent) -> void:
 
 func _on_return_button_press():
 	queue_free()
+	
+
+func _on_add_torch_pressed():
+	torch_confirmation_dialog.show()
+
+func _close_torch_placement_dialog():
+	torch_confirmation_dialog.hide()
+
+func _add_torch_to_current_location():
+	var current_player_floor_index = current_player_room.floor_index
+	var current_player_room_index = current_player_room.room_index
+	
+	var room_light_range = 1
+	var max_light_index = current_player_floor_index + _LIGHT_FLOOR_RANGE + 1 
```

Comment:
`max_light_height_index` ?
and why the +1 at the end ? if player is on floor 1, and the _LIGHT_FLOOR_RANGE is 1, it means the torch placed at level 1 is doing light until the level 2 (a range of 1)
unless you consider the current room as being the range of 1 instead of a range of 0 ?